modules/auxiliary/gather/ldap_query.rb

```diff
@@ -206,6 +206,20 @@ def output_data_csv(entries)
     generate_rex_tables(entries, 'csv')
   end
 
+  def normalize_entries(entries)
+    cleaned_entries = []
+    entries.each do |entry|
+      # Convert to a hash so we get only the data we need.
+      entry = entry.to_h
+      entry_keys = entry.keys
+      entry_keys.each do |key|
+        entry[key][0] = Rex::Text.to_hex_ascii(entry[key][0])
```

Comment:
This only applies the filtering to the first item in the array. It'll fail if the array is empty, but more importantly, it won't process any elements past the first one.

```suggestion
        entry[key] = entry[key].map { |v| Rex::Text.to_hex_ascii(v) }
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16955/commits/6b3d3913e761398226c33047babe42dcd4dfda27